xrpl/models/utils.py

```diff
@@ -1,9 +1,41 @@
 """Helper util functions for the models module."""
+
+import re
 from dataclasses import dataclass, is_dataclass
-from typing import Any, Dict, List, Type, TypeVar, cast
+from typing import Any, Dict, List, Optional, Pattern, Type, TypeVar, cast
+
+from typing_extensions import Final
 
 from xrpl.models.exceptions import XRPLModelException
 
+HEX_REGEX: Final[Pattern[str]] = re.compile("[a-fA-F0-9]*")
+
+# Credentials are represented in hex. Whilst they are allowed a maximum length of 64
+# bytes, every byte requires 2 hex characters for representation
+_MAX_CREDENTIAL_LENGTH: Final[int] = 64 * 2
+
+
+def validate_credential_ids(credential_list: Optional[List[str]]) -> Dict[str, str]:
+    """
+    Returns:
+        Errors pertaining to credential_ids field
+
+    Args:
+        credential_list: An optional list of input credentials
+
+    """
+    errors = {}
+    if credential_list is not None:
```

Comment:
```suggestion
    errors = {}
    if credential_list is None:
        return errors
```

Reply:
does it really make a difference? given that CPU instructions are re-ordered and if-else conditions are speculatively executed?

Reply:
It doesn't make a difference for efficiency, but it makes the code easier to read since it requires less indenting.

Reply:
But the current version of the code makes it easier to understand the business logic:

if the credential_list is not empty, then perform some validation checks
else don't worry about it

Reply:
Exiting early when you can is generally considered better practice. It might not matter too much now, but if more `if`s get added, the indents will start to build up, which is why short-circuiting like this is easier to read.

In other words, "if the credential list is empty, don't worry about the rest of the code in this function. Otherwise, you can read it".

Reply:
0f7f745d